core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/batchutilities/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.batchutilities;
+
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configuration values for the {@link BatchManager}. All values are optional, and the default values will be used
+ * if they are not specified.
+ */
+// TODO should this implement toCopyableBuilder?
+@SdkPublicApi
+public class BatchOverrideConfiguration implements ToCopyableBuilder<BatchOverrideConfiguration.Builder,
```

Comment:
Can we add `final` to all classes to prohibit inheritance since they are not intended to be subclassed?

Reply:
Made classes final